src/main/java/Parser.java

```diff
@@ -0,0 +1,45 @@
+import java.io.IOException;
+public class Parser {
+    public static boolean parse(String str, TaskList taskList, Ui ui, Storage storage) throws DukeException {
```

Comment:
Perhaps to make the method name shorter, you could consider wrapping the whole parse method in a try-catch? So you don't have to explicitly mention that it throws DukeException sooo something like

```
public static boolean parse(String str, TaskList taskList, Ui ui, Storage storage) { 
    try { //main logic 
     
    } catch (DukeException d) {
    // handle the exception here 
    } catch (moreExceptionsHere m) { 
    // gives you the freedom to handle more exceptions down the line 
```
Edit* : Sorry I just realized you were alr doing this i think can remove the `throws dukeException` from the code and it'll be okay alr